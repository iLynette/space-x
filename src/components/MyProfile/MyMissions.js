import styles from './Profile.module.css';

const MyMissions = () => (
  <>
    <ul className={styles['my-journeys']}>
      <li>
        {' '}
        <h1>My Missions</h1>
      </li>
      <li>Falcon 9</li>
      <li>Falcon Heavy</li>
      <li>Starship</li>
    </ul>
  </>
);

export default MyMissions;
