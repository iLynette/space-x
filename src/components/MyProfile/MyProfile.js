import React from 'react';
import { useSelector } from 'react-redux';

export default function MyProfile() {
  const dragons = useSelector((state) => state.dragonsReducer);
  const rockets = useSelector((state) => state.rocketReducer);
  const reservedRockets = rockets.filter((rocket) => rocket.rocketReserved);

  return (
    <div className="profileBody">
      <div>
        <h2 className="">My Missions</h2>
        <div className="profileMissions">
          {/* list of missions */}
          {/* <img src="https://img.icons8.com/external-tone-royyan-wijaya/64/000000/external-flying-space-duotone-tone-royyan-wijaya.png"/> */}
        </div>
      </div>
      <div>
        <h2 className="">My Rockets</h2>
        <div className="profileRockets">
        <ul className="reserve-rocket">
            {(reservedRockets.length === 0)? (
              <li className="no-reserved-rocket">
                No Rockets Reserved
              </li>
            ) : (
              reservedRockets
                .map((rocket) => (
                  <li key={rocket.rocketId} className="reserved-item">
                    <span className="name">{rocket.rocketName}</span>
                  </li>
                ))
            )}
          </ul>
        </div>
      </div>
      <div>
        <h2 className="">My Dragons</h2>
        <div className="profileDragons">
          {
          dragons.map((dragon) => (
            <>
              {dragon.reserved && (
                <div key={dragon.id} className="profileCard">
                  <img src="https://img.icons8.com/external-xnimrodx-blue-xnimrodx/64/000000/external-rocket-online-marketing-xnimrodx-blue-xnimrodx.png" alt="reserved" />
                  <h3>{dragon.name}</h3>
                </div>
              )}
            </>
          ))
          }
        </div>
      </div>
    </div>
  );
}
