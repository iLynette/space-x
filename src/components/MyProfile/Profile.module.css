.container {
  display: flex;
  justify-content: center;
}

.my-journeys {
  margin: 1rem;
  display: flex;
  flex-direction: column;
  width: 25vw;
}

.my-journeys li {
  padding: 1rem;
}

.my-journeys li:nth-child(2) {
  border-top-left-radius: 0.3rem;
  border-top-right-radius: 0.3rem;
}

.my-journeys li:not(:first-child) {
  border-top: 1px solid #dce1e5;
  border-left: 1px solid #dce1e5;
  border-right: 1px solid #dce1e5;
}

.my-journeys li:last-child {
  border-bottom: 1px solid #dce1e5;
  border-bottom-left-radius: 0.3rem;
  border-bottom-right-radius: 0.3rem;
}
