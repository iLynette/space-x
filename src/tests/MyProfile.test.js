import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile/MyProfile';

it('Test profile', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MyProfile />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
