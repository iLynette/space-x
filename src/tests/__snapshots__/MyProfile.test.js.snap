// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test profile 1`] = `
<div
  className="profileBody"
>
  <div>
    <h2
      className="reserve"
    >
      My Missions
    </h2>
    <div
      className="profileMissions"
    >
      <ul>
        <li
          className="no-reserved-mission"
        >
          No Mission Reserved
        </li>
      </ul>
    </div>
  </div>
  <div>
    <h2
      className=""
    >
      My Rockets
    </h2>
    <div
      className="profileRockets"
    >
      <ul
        className="reserve-rocket"
      >
        <li
          className="no-reserved-rocket"
        >
          No Rockets Reserved
        </li>
      </ul>
    </div>
  </div>
  <div>
    <h2
      className=""
    >
      My Dragons
    </h2>
    <div
      className="profileDragons"
    >
      <ul>
        <li
          className="profileCard"
        >
          <p>
            No Dragons Reserved
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
